contrib/mongo-c-driver-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,280 @@
+option(USE_MONGODB "Enable MongoDB" ${ENABLE_LIBRARIES})
+
+if (NOT USE_MONGODB)
+    message(STATUS "Not using libmongoc and libbson")
+    return()
+endif()
+
+set (libbson_VERSION_MAJOR 1)
+set (libbson_VERSION_MINOR 27)
+set (libbson_VERSION_PATCH 0)
+set (libbson_VERSION 1.27.0)
+set (libmongoc_VERSION_MAJOR 1)
+set (libmongoc_VERSION_MINOR 27)
+set (libmongoc_VERSION_PATCH 0)
+set (libmongoc_VERSION 1.27.0)
+
+set(LIBBSON_SOURCES_ROOT "${ClickHouse_SOURCE_DIR}/contrib/mongo-c-driver/src")
+set(LIBBSON_SOURCE_DIR "${LIBBSON_SOURCES_ROOT}/libbson/src")
+file(GLOB_RECURSE LIBBSON_SOURCES "${LIBBSON_SOURCE_DIR}/*.c")
+
+include (TestBigEndian)
+test_big_endian (BSON_BIG_ENDIAN)
+if (BSON_BIG_ENDIAN)
+    set (BSON_BYTE_ORDER 4321)
+else ()
+    set (BSON_BYTE_ORDER 1234)
+endif ()
+
+include (CheckFunctionExists)
```

Comment:
This is strictly forbidden :vomiting_face:

Reply:
it was a draft :)
fixed.